bundles/framework/publisher2/view/PublisherSidebar.js

```diff
@@ -518,6 +518,11 @@ Oskari.clazz.define('Oskari.mapframework.bundle.publisher2.view.PublisherSidebar
                 var event = Oskari.eventBuilder('Publisher2.ToolEnabledChangedEvent')(tool);
                 sb.notifyAll(event);
             });
+
+            // fix layer order after changing from/to publisher
+            setTimeout(() => {
+                sb.findRegisteredModuleInstance('MainMapModule').ensureLayerOrder();
```

Comment:
Does this really need the timeout? Here's a function that also does something with the layers so maybe move the ordering stuff here https://github.com/oskariorg/oskari-frontend/blob/179971a84ec2f93c10c3d70c1bbf20138f2e6346/bundles/framework/publisher2/instance.js#L267 or even publisher2/service/PublisherService.js

What makes the layers go out of order?